vault/raft.go

```diff
@@ -160,11 +122,24 @@ func (c *Core) startRaftBackend(ctx context.Context) (retErr error) {
 		return nil
 	}
 
+	raftBackend.SetFollowerStates(c.raftFollowerStates)
 	raftBackend.SetRestoreCallback(c.raftSnapshotRestoreCallback(true, true))
+
+	// Read autopilot configuration from storage
+	autopilotConfig, err := c.autopilotConfiguration(ctx)
+	if err != nil {
+		c.logger.Error("failed to load autopilot config from storage when setting up cluster; continuing since autopilot falls back to default config", "error", err)
+	}
+
+	if c.disableAutopilot {
+		raftBackend.DisableAutopilot()
+	}
+
 	if err := raftBackend.SetupCluster(ctx, raft.SetupOpts{
 		TLSKeyring:      raftTLS,
 		ClusterListener: c.getClusterListener(),
 		StartAsLeader:   creating,
+		AutopilotConfig: autopilotConfig,
```

Comment:
I think there is some kind of bug here:

Read Config on leader node (default config):
```
% vault2 read sys/storage/raft/autopilot/configuration
Key                                   Value
---                                   -----
cleanup_dead_servers                  false
dead_server_last_contact_threshold    24h0m0s
last_contact_threshold                10s
max_trailing_logs                     1000
min_quorum                            3
server_stabilization_time             10s
```

Update Config:
```
% vault2 write sys/storage/raft/autopilot/configuration cleanup_dead_servers=true min_quorum=3
Success! Data written to: sys/storage/raft/autopilot/configuration
```

Read config again (works fine):
```
% vault2 read sys/storage/raft/autopilot/configuration
Key                                   Value
---                                   -----
cleanup_dead_servers                  true
dead_server_last_contact_threshold    24h0m0s
last_contact_threshold                10s
max_trailing_logs                     1000
min_quorum                            3
server_stabilization_time             10s
```

Stepdown as leader:
```
% vault2 operator step-down
Success! Stepped down: http://127.0.0.2:8200
```

Read config on new leader (reverted back to default config):
```
% vault4 read sys/storage/raft/autopilot/configuration
Key                                   Value
---                                   -----
cleanup_dead_servers                  false
dead_server_last_contact_threshold    24h0m0s
last_contact_threshold                10s
max_trailing_logs                     1000
min_quorum                            3
server_stabilization_time             10s
```

Reply:
It might be that we need to re-load the config in `setupRaftActiveNode` when the node becomes active?

Reply:
I had thought of doing that. It must have slipped my mind. Done. Fixed.